### PR TITLE
Add options-based command syntax support

### LINKED ISSUE #147: Update command syntax to accept index-based parameters
E.g.:
- `edit -p PATIENT_INDEX -t TASK_INDEX` as a shorthand for `editTask`;
- `edit -p PATIENT_INDEX` or `edit PATIENT_INDEX` as shorthand for `editPatient`.

### PR DESCRIPTION
**Pull request by rama-pang**
Resolves #147.

`ParseUtil` now supports parsing `-[option]` command options. We assume that command options never use the character `/`, and they are written before text-based parameters (i.e. `[descriptor]/` parameters).

Additionally, when a user enters a command, two or more consecutive white spaces will be treated as a single white space. E.g., <pre>find  one two   three</pre> will be treated as <pre>find one two three</pre>

Changes the following commands:
- Allow user to call command `add -p PATIENT_INDEX [d/TASK_DESCRIPTION]` to add a task to patient with index `PATIENT_INDEX`. User no longer able to use `addTask` command.
-  `AddPatientCommand` and `AddTaskCommand` now inherits from `AddGenericCommand`. `COMMAND_WORD` is deleted from `AddPatientComand` and `AddTaskCommand`, as they are no longer used.
- Allow user to call command `edit -p PATIENT_INDEX` to edit patient with index `PATIENT_INDEX`. User no longer able to use `editPatient` command.
- Allow user to call command `edit -p PATIENT_INDEX -t TASK_INDEX [d/TASK_DESCRIPTION]` to edit task `TASK_INDEX` of patient with index `PATIENT_INDEX`. User no longer able to use `editTask` command.
-  `EditPatientCommand` and `EditTaskCommand` now inherits from `EditGenericCommand`. `COMMAND_WORD` is deleted from `EditPatientComand` and `EditTaskCommand`, as they are no longer used.
- Allow user to call command `delete -p PATIENT_INDEX` to delete patient with index `PATIENT_INDEX`. User no longer able to use `deletePatient` command.
- Allow user to call command `delete -p PATIENT_INDEX -t TASK_INDEX` to delete task `TASK_INDEX` of patient with index `PATIENT_INDEX`. User no longer able to use `deleteTask` command.
-  `DeletePatientCommand` and `DeleteTaskCommand` now inherits from `DeleteGenericCommand`. `COMMAND_WORD` is deleted from `DeletePatientComand` and `DeleteTaskCommand`, as they are no longer used.

To dos:
- Add documentations.
- Fix message errors

### CODE REVIEW

**Review by izzahaj — COMMENTED** [review]
LGTM, though one other person should have a look as well.

**Review by Ramanathan0908 — APPROVED** [review]
LGTM

**Comment by rama-pang** [comment]
Perhaps this PR should wait until #148, #112, and #111 are merged, since it is a breaking change.

File: src/main/java/seedu/address/logic/parser/EditGenericCommandParser.java
Comment (izzahaj) [inline src/main/java/seedu/address/logic/parser/EditGenericCommandParser.java:28]: A code quality suggestion: would it be better to evaluate `options.getValue(PREFIX_OPTION_PATIENT_INDEX).isPresent()` and  `options.getValue(PREFIX_OPTION_TASK_INDEX).isPresent()` first as an intermediate value so the condition doesn't look too long? Perhaps just `options.getValue(PREFIX_OPTION_PATIENT_INDEX)` and `options.getValue(PREFIX_OPTION_TASK_INDEX)` since they are used multiple times with `get()` too?

**Review by izzahaj — APPROVED** [review]
LGTM, though some conditions in the `EditGenericCommandParser`, `DeleteGenericCommandParser` and `AddGenericCommandParser` can be replaced with intermediate values to improve readability.

File: src/main/java/seedu/address/logic/parser/EditGenericCommandParser.java
Comment (rama-pang) [reply src/main/java/seedu/address/logic/parser/EditGenericCommandParser.java]: Done

**Pull request merged**
